code/API_definitions/qod-api.yaml

```diff
@@ -325,20 +331,52 @@ components:
         * `QOS_S` - Qualifier for the requested QoS profile _S_
         * `QOS_M` - Qualifier for the requested QoS profile _M_
         * `QOS_L` - Qualifier for the requested QoS profile _L_
+    
     Notification:
       type: object
       required:
         - sessionId
-        - event
+        - eventReports
       properties:
         sessionId:
           $ref: "#/components/schemas/SessionId"
-        event:
-          $ref: "#/components/schemas/SessionEvent"
-    SessionEvent:
+        eventReports:
+          type: array
+          items:
+            $ref: "#/components/schemas/EventReport"
+          minItems: 1
+          description: Contains the reported event and applicable information
+    
+    EventReport:
+      oneOf:
+        - $ref: '#/components/schemas/QosStatusChangedEventReport'
+      discriminator:
+        propertyName: eventType
+        mapping:
+          QOS_STATUS_CHANGED: "#/components/schemas/QosStatusChangedEventReport"
+        
+    QosStatusChangedEventReport:
+      type: object
+      required:
+        - eventType
+        - qosStatus
+      properties:
+        eventType:
+          $ref: "#/components/schemas/SessionEventType"
+        qosStatus:
+          $ref: "#/components/schemas/QosStatus"
+        statusInfo:
+          type: string
+          enum:
+            - DURATION_EXPIRED
+            - NETWORK_TERMINATED
+          description: Currently statusInfo is only applicable when qosStatus is 'UNAVAILABLE'.
```

Comment:
With this proposal we are assuming that once QoS becomes UNAVAILABLE (due to duration expires or for network reasons), the session is implicitly terminated. I'm wondering if that will be always the case, or there may be situations where QoS is temporarily unavailable or degraded, but session is still active, so QoS could be eventually restored. Is this scenario possible?